projects/core/src/configurator/commons/store/actions/configurator.action.ts

```diff
@@ -34,136 +36,161 @@ export const UPDATE_PRICE_SUMMARY_FAIL =
   readonly type = UPDATE_CONFIGURATION_FINALIZE_FAIL;
   constructor(public payload: Configurator.Configuration) {
-    super(CONFIGURATION_DATA, payload.productCode);
+    super(CONFIGURATION_DATA, payload.owner.key);
   }
 }
 
 export class UpdatePriceSummary extends StateEntityLoaderActions.EntityLoadAction {
   readonly type = UPDATE_PRICE_SUMMARY;
   constructor(public payload: Configurator.Configuration) {
-    super(CONFIGURATION_DATA, payload.productCode);
+    super(CONFIGURATION_DATA, payload.owner.key);
   }
 }
 export class UpdatePriceSummaryFail extends StateEntityLoaderActions.EntityFailAction {
   readonly type = UPDATE_PRICE_SUMMARY_FAIL;
-  constructor(productCode: string, public payload: any) {
-    super(CONFIGURATION_DATA, productCode, payload);
+  constructor(ownerKey: string, public payload: any) {
+    super(CONFIGURATION_DATA, ownerKey, payload);
   }
 }
 
 export class UpdatePriceSummarySuccess extends StateEntityLoaderActions.EntitySuccessAction {
   readonly type = UPDATE_PRICE_SUMMARY_SUCCESS;
   constructor(public payload: Configurator.Configuration) {
-    super(CONFIGURATION_DATA, payload.productCode);
+    super(CONFIGURATION_DATA, payload.owner.key);
   }
 }
 
 export class ChangeGroup extends StateEntityLoaderActions.EntityLoadAction {
   readonly type = CHANGE_GROUP;
-  constructor(public payload: any) {
-    super(CONFIGURATION_DATA, payload.productCode);
+  constructor(
+    public configuration: Configurator.Configuration,
+    public groupId: string
+  ) {
+    super(CONFIGURATION_DATA, configuration.owner.key);
   }
 }
 
 export class ChangeGroupFinalize extends StateEntityLoaderActions.EntityLoadAction {
   readonly type = CHANGE_GROUP_FINALIZE;
-  constructor(public payload: any) {
-    super(CONFIGURATION_DATA, payload.productCode);
+  constructor(public payload: Configurator.Configuration) {
+    super(CONFIGURATION_DATA, payload.owner.key);
   }
 }
 
 export class AddToCart extends StateEntityLoaderActions.EntityLoadAction {
   readonly type = ADD_TO_CART;
-  constructor(public payload: any) {
+  constructor(public payload: Configurator.AddToCartParameters) {
     super(CONFIGURATION_DATA, payload.productCode);
   }
 }
 
-export class AddToCartFinalize extends StateEntityLoaderActions.EntityLoadAction {
-  readonly type = ADD_TO_CART_FINALIZE;
-  constructor(public payload: any) {
-    super(CONFIGURATION_DATA, payload.productCode);
+export class RemoveConfiguration extends StateEntityLoaderActions.EntityResetAction {
+  readonly type = REMOVE_CONFIGURATION;
+  constructor(ownerKey: string | string[]) {
+    super(CONFIGURATION_DATA, ownerKey);
   }
 }
 
-export class RemoveConfiguration extends StateEntityLoaderActions.EntityResetAction {
-  readonly type = REMOVE_CONFIGURATION;
-  constructor(productCode: string | string[]) {
-    super(CONFIGURATION_DATA, productCode);
+export class AddOwner extends StateEntityLoaderActions.EntityLoadAction {
```

Comment:
Not a loader action, can be an entity action

Reply:
Done (rather a pure action instead of entity action)